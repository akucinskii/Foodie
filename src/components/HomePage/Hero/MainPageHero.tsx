import { signIn, useSession } from "next-auth/react";
import Link from "next/link";
import React from "react";
import Button from "../../Button/Button";

const MainPageHero = () => {
  const { data: session } = useSession();

  return (
    <div
      className=" hero z-10 min-h-screen w-full bg-gradient-to-br
    from-secondary to-primary"
    >
      <div className="hero-content col-start-1 row-start-1 w-full max-w-7xl flex-col justify-between gap-10 pb-40 text-center lg:gap-0 xl:gap-20">
        <div className="max-w-2xl">
          <h1 className="text-5xl font-bold" data-testid="title-header">
            Welcome to solve
            <span className="font-extrabold text-yellow-500">M</span>c
          </h1>
          <p className="font-title mt-5 text-xl italic">
            We make ordering food with friends easier, faster and more fun!
          </p>
          <div className="mt-5 flex justify-center gap-2">
            <Link href="/restaurant/restaurants">
              <button
                data-testid="link-to-all"
                className="
                btn
              btn-active
              btn-ghost
              text-white
              lg:btn-lg
              "
              >
                See all restaurants
              </button>
            </Link>
            {!session && (
              <Button
                data-testid="signin-button"
                className="
            btn
            btn-primary
            text-white
            lg:btn-lg
            "
                onClick={() => {
                  signIn();
                }}
              >
                Sign In
              </Button>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default MainPageHero;
