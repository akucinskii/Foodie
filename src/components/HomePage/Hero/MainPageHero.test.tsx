import { fireEvent, getByTestId, render } from "@testing-library/react";
import { Session } from "next-auth";
import { signIn, signOut, useSession } from "next-auth/react";

import { afterAll, describe, expect, Mock, test, vi } from "vitest";

import MainPageHero from "./MainPageHero";

const mockSession: Session = {
  user: {
    id: "1",
    name: "Test",
    email: "test@test.test",
    image: "https://testimages.org/img/testimages_screenshot.jpg",
  },
  expires: "1",
};

vi.mock("next-auth/react", () => ({
  signIn: vi.fn(),
  signOut: vi.fn(),
  useSession: vi.fn(),
}));

vi.mock("next/router", () => ({
  useRouter: vi.fn(() => ({
    router: {
      push: vi.fn(),
    },
  })),
}));

describe("Button", () => {
  (signIn as Mock).mockImplementation(() => vi.fn());
  (signOut as Mock).mockImplementation(() => vi.fn());

  (useSession as Mock).mockReturnValue({
    data: mockSession,
    status: "authenticated",
  });

  const { container, rerender } = render(<MainPageHero />);

  test("should match snapshot", () => {
    (useSession as Mock).mockReturnValue({
      data: mockSession,
      status: "authenticated",
    });
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot when logout", () => {
    (useSession as Mock).mockReturnValue({
      data: mockSession,
      status: "authenticated",
    });

    rerender(<MainPageHero />);
    expect(container).toMatchSnapshot("logout");
  });

  test("Should render sign in button", () => {
    (useSession as Mock).mockReturnValue({
      data: null,
      status: "unauthenticated",
    });
    rerender(<MainPageHero />);
    expect(container);
  });

  test("should call signIn when click on login button", async () => {
    (useSession as Mock).mockImplementation(() => ({
      data: null,
      status: "unauthenticated",
    }));

    rerender(<MainPageHero />);

    fireEvent.click(getByTestId(container, "button"));

    expect(signIn).toHaveBeenCalled();
  });

  afterAll(() => {
    (signIn as Mock).mockRestore();
    (signOut as Mock).mockRestore();
    (useSession as Mock).mockRestore();
  });
});
