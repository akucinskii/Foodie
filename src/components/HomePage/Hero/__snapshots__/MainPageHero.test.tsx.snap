// Vitest Snapshot v1

exports[`Button > should match snapshot 1`] = `
<div>
  <div
    class=" hero z-10 min-h-screen w-full bg-gradient-to-br from-secondary to-primary"
  >
    <div
      class="hero-content col-start-1 row-start-1 w-full max-w-7xl flex-col justify-between gap-10 pb-40 text-center lg:gap-0 xl:gap-20"
    >
      <div
        class="max-w-2xl"
      >
        <h1
          class="text-5xl font-bold"
          data-testid="title-header"
        >
          Welcome to solve
          <span
            class="font-extrabold text-yellow-500"
          >
            M
          </span>
          c
        </h1>
        <p
          class="font-title mt-5 text-xl italic"
        >
          We make ordering food with friends easier, faster and more fun!
        </p>
        <div
          class="mt-5 flex justify-center gap-2"
        >
          <button
            class=" btn btn-active btn-ghost text-white lg:btn-lg "
            data-testid="link-to-all"
          >
            See all restaurants
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Button > should match snapshot when logout > logout 1`] = `
<div>
  <div
    class=" hero z-10 min-h-screen w-full bg-gradient-to-br from-secondary to-primary"
  >
    <div
      class="hero-content col-start-1 row-start-1 w-full max-w-7xl flex-col justify-between gap-10 pb-40 text-center lg:gap-0 xl:gap-20"
    >
      <div
        class="max-w-2xl"
      >
        <h1
          class="text-5xl font-bold"
          data-testid="title-header"
        >
          Welcome to solve
          <span
            class="font-extrabold text-yellow-500"
          >
            M
          </span>
          c
        </h1>
        <p
          class="font-title mt-5 text-xl italic"
        >
          We make ordering food with friends easier, faster and more fun!
        </p>
        <div
          class="mt-5 flex justify-center gap-2"
        >
          <button
            class=" btn btn-active btn-ghost text-white lg:btn-lg "
            data-testid="link-to-all"
          >
            See all restaurants
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
